api/controllers/zetter.ts

```diff
@@ -1,294 +1,328 @@
-    tweet['numberOfFavorite'] = numberOfFavorite.length;
-
-    const numberOfReply: Array<TweetType> = tweets.filter((replyTweet) => replyTweet.replyTo === tweet.id);
-    tweet['numberOfReply'] = numberOfReply.length;
-
-    // ツイートに自分がfavしているかの情報を付加するための準備
-    const favoriteTweetIds: Array<number> = favorities
-        .filter((favorite) => favorite.userId === req.user.id)
-        .map((obj) => obj.tweetId);
-    tweet.isFavorite = favoriteTweetIds.includes(tweet.id);
-
     res.json(tweet);
 };
 
-export const getReplys = (req: Request, res: Response) => {
-    let replys: Array<TweetType> = [];
-    tweets.forEach((tweet) => {
-        if (tweet.replyTo === parseInt(req.params.tweetId)) {
-            replys.push(tweet);
-        }
+export const getReplys = async (req: Request, res: Response) => {
+    const replys = await prisma.tweet.findMany({
+        where: {
+            replyTo: parseInt(req.params.tweetId),
+        },
+        include: { user: true, favorities: true },
     });
 
-    const result: Array<TweetType> = addInformationToTweet(replys, req);
-    res.json(result);
+    res.json(replys);
 };
 
-export const createTweet = (req: Request, res: Response) => {
-    const newTweet: TweetType = {
-        id: tweets.length + 1,
-        createdBy: req.user.id,
-        replyTo: req.body.replyTo,
-        content: req.body.content,
-        createdAt: new Date(),
-    };
-    tweets.push(newTweet);
+export const createTweet = async (req: Request, res: Response) => {
+    const newTweet: Tweet = await prisma.tweet.create({
+        data: {
+            createdBy: req.user.id,
+            replyTo: req.body.replyTo,
+            content: req.body.content,
+            createdAt: new Date(),
+        },
+    });
     res.status(201).json(newTweet);
 };
 
-export const updateTweet = (req: Request, res: Response) => {
-    const tweet: TweetType = req.body.tweet;
-    if (!tweet.numberOfFavorite) {
-        tweet.numberOfFavorite = 0;
+export const updateTweet = async (req: Request, res: Response) => {
+    const tweet: Tweet | null = await prisma.tweet.findUnique({
+        where: {
+            id: req.body.tweet.id,
+        },
+        include: {
+            user: true,
+            favorities: true,
+        },
+    });
+
+    if (!tweet) {
+        throw Error;
     }
 
     if (req.body.order === 'add') {
-        favorities.push({
-            id: favorities.length + 1,
-            // TODO: DBとつなぐときなおしたい
-            tweetId: tweet.id,
-            userId: req.user.id,
-            createdAt: new Date(),
+        await prisma.favorite.create({
+            data: {
+                tweetId: tweet.id,
+                userId: req.user.id,
+            },
+        });
+
+        await prisma.tweet.update({
+            where: {
+                id: tweet.id,
+            },
+            data: {
+                numberOfFavorite: tweet.numberOfFavorite + 1,
+            },
```

Comment:
これは一般的な知識というよりprismaの特殊なメソッドやけど、こういうのものある
https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#atomic-number-operations

このメソッドを使うと、tweetを`findUnique`してから`update`するまでの間に他のプロセスが`numberOfFavorite`をいじってしまったケースにも安心になる

```suggestion
        await prisma.tweet.update({
            where: {
                id: tweet.id,
            },
            data: {
                numberOfFavorite: { increment: 1 }
            },
```